cmd/api/app/routes/unstructured/handler.go

```diff
@@ -39,7 +39,8 @@ func handleDeleteResource(c *gin.Context) {
 	}
 	kind := c.Param("kind")
 	namespace := c.Param("namespace")
-	if namespace == "" {
+	// if `kind` is 'namespace', should not set `namespace=default`
+	if namespace == "" && kind != "namespace" {
```

Comment:
Is this function `handleDeleteResource` used by all resource types? Especially cluster-scoped resources other than `namespace`.

Reply:
Only one cluster wide resource is namespace resource, I got what you concerned, I should extract a method to check whether the resource is cluster wide or namespace scoped, for all kinds of cluster wide resource, it should not set default value for namespace

Reply:
> Only one cluster scoped resource is namespace resource currently.

I haven't tested the `ClusterPropagationPolicy` and `ClusterOverridePolicy`, but they are both cluster-scoped resources.

> I should extract a method to check whether the resource is cluster scoped or namespace scoped, for all kinds of cluster scoped resource, it should not set default value for namespace

Not sure about it. I'm not sure if it is necessary to set the namespace to `default`.

Reply:
I just reviewed the code, `ClusterPropagationPolicy` and `ClusterOverridePolicy` will not delete by `handleDeleteResource` function, they are handled in another two functions.
for `ClusterPropagationPolicy` resource, it will be delete in the following function:
https://github.com/karmada-io/dashboard/blob/be1beb1c272ecdd88b8273a88a8aa58ed95a37a7/cmd/api/app/routes/propagationpolicy/handler.go#L141-L176

for `ClusterOverridePolicy` resource, , it will be delete in the following function:
https://github.com/karmada-io/dashboard/blob/be1beb1c272ecdd88b8273a88a8aa58ed95a37a7/cmd/api/app/routes/overridepolicy/handler.go#L141-L176

Reply:
> Not sure about it. I'm not sure if it is necessary to set the namespace to default.

I think It's a kind of guard mechanism.  Just like kubectl apply, if the `namespace` is not set, the kubectl will assumed that the `namespace` is default for namespace-scoped resources like `deployments`, but will not assumed that `namespace` for cluster-scoped resources.

Reply:
removed the logic of setting default value for `namespace`, and checked the delete for namespace-scoped resource locally